db/model/roomType.js

```diff
@@ -69,13 +74,170 @@ module.exports = function roomType(seq, dataTypes) {
         assoc.type = RoomType.hasMany(models.Room, {
           foreignKey: 'type',
         });
+        assoc.bots = RoomType.belongsToMany(models.Bot, {
+          foreignKey: 'roomTypeId',
+          through: 'RoomTypeBots',
+        });
         assoc.writers = RoomType.belongsToMany(models.User, {
           as: 'writers',
           through: 'RoomTypeWriters',
           foreignKey: 'roomTypeId',
         });
       },
     },
+    hooks: {
+
+      /**
+       * Ensures that all bots in request actually exist
+       *
+       * @param {Instance} inst - The instance being created
+       * @returns {Promise} which resolves to the instance, or rejects if
+       *  a bot is not found or duplicate bots are requested
+       */
+      beforeCreate(inst /* , opts */) {
+        const bots = inst.dataValues.bots;
+
+        return new seq.Promise((resolve, reject) => {
+          if (bots == null) {
+            resolve(inst);
+          }
+
+          if (bots.length > new Set(bots).size) {
+            reject(new Error(`Cannot have duplicate bots`));
+          }
+
+          bots.map((botName, index) => {
```

Comment:
using .map(...) suggests you're returning an array, so if you're *not* returning an array and just iterating over the elements in the array, prefer to use .forEach for that